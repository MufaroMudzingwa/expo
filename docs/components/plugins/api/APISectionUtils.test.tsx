import { render } from '@testing-library/react';
import * as React from 'react';

import { resolveTypeName } from './APISectionUtils';

describe('APISectionUtils.resolveTypeName', () => {
  test('void', () => {
    const { container } = render(<>{resolveTypeName({ type: 'intrinsic', name: 'void' })}</>);
    expect(container).toMatchSnapshot();
  });

  test('generic type', () => {
    const { container } = render(<>{resolveTypeName({ type: 'intrinsic', name: 'string' })}</>);
    expect(container).toMatchSnapshot();
  });

  test('custom type', () => {
    const { container } = render(
      <>{resolveTypeName({ type: 'reference', name: 'SpeechSynthesisEvent' })}</>
    );
    expect(container).toMatchSnapshot();
  });

  test('custom type array', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'array',
          elementType: { type: 'reference', name: 'AppleAuthenticationScope' },
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });

  test('Promise', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'reference',
          typeArguments: [{ type: 'intrinsic', name: 'void' }],
          name: 'Promise',
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });

  test('Promise with custom type', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'reference',
          typeArguments: [{ type: 'reference', name: 'AppleAuthenticationCredential' }],
          name: 'Promise',
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });

  test('Record', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'reference',
          typeArguments: [
            { type: 'intrinsic', name: 'string' },
            { type: 'intrinsic', name: 'any' },
          ],
          name: 'Record',
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });

  test('union', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'union',
          types: [
            { type: 'reference', name: 'SpeechEventCallback' },
            { type: 'literal', value: null },
          ],
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });

  test('union with array', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'union',
          types: [
            { type: 'array', elementType: { type: 'intrinsic', name: 'number' } },
            { type: 'literal', value: null },
          ],
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });

  test('function', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'reflection',
          declaration: {
            signatures: [
              {
                type: {
                  type: 'union',
                  types: [
                    { type: 'intrinsic', name: 'void' },
                    {
                      type: 'reference',
                      name: 'SpeechEventCallback',
                    },
                  ],
                },
              },
            ],
          },
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });

  test('function with arguments', () => {
    const { container } = render(
      <>
        {resolveTypeName({
          type: 'reflection',
          declaration: {
            signatures: [
              {
                parameters: [
                  {
                    name: 'error',
                    type: { type: 'reference', name: 'Error' },
                  },
                ],
                type: {
                  type: 'union',
                  types: [
                    { type: 'intrinsic', name: 'void' },
                    { type: 'reference', name: 'SpeechEventCallback' },
                  ],
                },
              },
            ],
          },
        })}
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
