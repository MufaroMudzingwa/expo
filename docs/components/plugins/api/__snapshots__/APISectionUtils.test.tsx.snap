// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APISectionUtils.resolveTypeName Promise 1`] = `
<div>
  <span>
    Promise
    &lt;
    void
    &gt;
  </span>
</div>
`;

exports[`APISectionUtils.resolveTypeName Promise with custom type 1`] = `
<div>
  <span>
    Promise
    &lt;
    <a
      class="css-12zqqiz-STYLES_EXTERNAL_LINK"
      href="/#appleauthenticationcredential"
    >
      AppleAuthenticationCredential
    </a>
    &gt;
  </span>
</div>
`;

exports[`APISectionUtils.resolveTypeName Record 1`] = `
<div>
  <span>
    Record
    &lt;
    string,any
    &gt;
  </span>
</div>
`;

exports[`APISectionUtils.resolveTypeName custom type 1`] = `
<div>
  <a
    class="css-12zqqiz-STYLES_EXTERNAL_LINK"
    href="/#speechsynthesisevent"
  >
    SpeechSynthesisEvent
  </a>
</div>
`;

exports[`APISectionUtils.resolveTypeName custom type array 1`] = `
<div>
  <a
    class="css-12zqqiz-STYLES_EXTERNAL_LINK"
    href="/#appleauthenticationscope"
  >
    AppleAuthenticationScope
    []
  </a>
</div>
`;

exports[`APISectionUtils.resolveTypeName function 1`] = `
<div>
  () =&gt;
   
  <span>
    void
     | 
  </span>
  <span>
    <a
      class="css-12zqqiz-STYLES_EXTERNAL_LINK"
      href="/#speecheventcallback"
    >
      SpeechEventCallback
    </a>
  </span>
</div>
`;

exports[`APISectionUtils.resolveTypeName function with arguments 1`] = `
<div>
  (
  <span>
    error
    : 
    <a
      class="css-12zqqiz-STYLES_EXTERNAL_LINK"
      href="/#error"
    >
      Error
    </a>
  </span>
  ) 
  =&gt;
   
  <span>
    void
     | 
  </span>
  <span>
    <a
      class="css-12zqqiz-STYLES_EXTERNAL_LINK"
      href="/#speecheventcallback"
    >
      SpeechEventCallback
    </a>
  </span>
</div>
`;

exports[`APISectionUtils.resolveTypeName generic type 1`] = `
<div>
  string
</div>
`;

exports[`APISectionUtils.resolveTypeName union 1`] = `
<div>
  <span>
    <a
      class="css-12zqqiz-STYLES_EXTERNAL_LINK"
      href="/#speecheventcallback"
    >
      SpeechEventCallback
    </a>
     | 
  </span>
  <span>
    null
  </span>
</div>
`;

exports[`APISectionUtils.resolveTypeName union with array 1`] = `
<div>
  <span>
    number[]
     | 
  </span>
  <span>
    null
  </span>
</div>
`;

exports[`APISectionUtils.resolveTypeName void 1`] = `
<div>
  void
</div>
`;
